### PR TITLE
Fix Mono runtime failures.

### LINKED ISSUE #45: Version 0.8.2 breaks android support
Hello,

After I updated from 0.8.1 to 0.8.2, I started getting the following exception when trying to use the CompressionStream class:

```
System.TypeInitializationException: The type initializer for 'ZstdSharp.Unsafe.Methods' threw an exception.
 ---> System.PlatformNotSupportedException: Operation is not supported on this platform.
   at System.Runtime.CompilerServices.RuntimeHelpers.AllocateTypeAssociatedMemory(Type type, Int32 size)
   at ZstdSharp.Unsafe.Methods..cctor()
   --- End of inner exception stack trace ---
   at ZstdSharp.SafeCctxHandle.Create()
   at ZstdSharp.Compressor..ctor(Int32 level)
   at ZstdSharp.CompressionStream..ctor(Stream stream, Int32 level, Int32 bufferSize, Boolean leaveOpen)
```

### PR DESCRIPTION
**Pull request by teo-tsirpanis**
Fixes #45.

### CODE REVIEW

File: src/ZstdSharp/Unsafe/Bitstream.cs [hunk 1]
```diff
@@ -10,7 +10,7 @@ namespace ZstdSharp.Unsafe
 {
     public static unsafe partial class Methods
     {
-#if NET8_0_OR_GREATER
+#if NET7_0_OR_GREATER
```
Comment (oleg-st) [inline src/ZstdSharp/Unsafe/Bitstream.cs:13]: As far as I know this optimization only works for `byte` / `sbyte` / `bool` in .NET 7 and extended in .NET 8
This part of code is made by [ZstdSharp.Generator](https://github.com/oleg-st/ZstdSharp.Generator)
See
https://github.com/oleg-st/ZstdSharp.Generator/blob/b7810ee1b65ef90b39a63caa3833b463bb69540d/src/ZstdSharp.Generator/CodeGenerator/CodeGenerator.Helper.cs#L743

**Review by oleg-st — CHANGES_REQUESTED** [review]

**Comment by oleg-st** [comment]
See
https://godbolt.org/z/Yr4njv9Wr

File: src/ZstdSharp/Unsafe/Bitstream.cs (hunk 1 above)
Comment (teo-tsirpanis) [reply src/ZstdSharp/Unsafe/Bitstream.cs:13]: The required [runtime API](https://learn.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.runtimehelpers.createspan) was added in .NET 7, but compiler support was added in the .NET 8 SDK. I built the .NET 7 assembly locally with the .NET 9 SDK and it produced the optimized code.

image

And that's how the code for this library was generated, awesome!

**Comment by oleg-st** [comment]
@teo-tsirpanis Thank you

**Pull request merged**